### PR TITLE
Update docs on address structure

### PR DESCRIPTION
**Pull request by flyingleafe**
Update documentation on address structure.

WIP, because this docs won't be completely up-to-date, unless we rewrite everything about binary serialization and synchronize the docs with current do

### CODE REVIEW

File: _docs/technical/protocols/2017-01-13-binary.md [hunk 1]
```diff
@@ -245,10 +245,10 @@ newtype Coin = Coin
 ```
 
 [Source code
-link](https://github.com/input-output-hk/cardano-sl/blob/1239a6a876fbbb970e836c8bc6bb72d781ba2fa9/core/Pos/Core/Types.hs#L271).
```
Comment (Martoon-00) [inline _docs/technical/protocols/2017-01-13-binary.md:248]: Ahah, nice link it used to be

File: _docs/technical/protocols/2017-01-13-binary.md [hunk 2]
```diff
@@ -559,48 +573,182 @@ Example:
...
+[Source code
+link](https://github.com/input-output-hk/cardano-sl/blob/524bb4d6d6620bec7e269e33de1154d2b1be5183/core/Pos/Core/Types.hs#L199)
 
+Address attributes hold additional information about address:
+
+- encrypted derivation path from the root of HD wallet tree (`aaPkDerivationPath`)
+- stake distribution associated with this address (`aaStakeDistribution`)
+
+Note that address may not contain an encrypted derivation path. Such addresses
+are are usually derived from level 0 (wallet) or level 1 (account) public keys
```
Comment (Martoon-00) [inline _docs/technical/protocols/2017-01-13-binary.md]: Typo
> are are

Also, it's not clear at first whether do you mean addresses w/o or w/ derivation path

File: _docs/technical/protocols/2017-01-13-binary.md [hunk 3]
```diff
@@ -559,48 +573,182 @@ Example:
...
+    -- 'SingleKeyDistr' can be used instead (which is smaller).
+    deriving (Eq, Ord, Show, Generic, Typeable)
+```
+
+[Source code
+link](https://github.com/input-output-hk/cardano-sl/blob/524bb4d6d6620bec7e269e33de1154d2b1be5183/core/Pos/Core/Types.hs#L137)
+
+In Byron era, all addresses should use `BootstrapEraDistr` as stake
+distribution -- which means, all the stake is automatically delegated to
+bootstrap stakeholders.
```
Comment (Martoon-00) [inline _docs/technical/protocols/2017-01-13-binary.md:651]: You just intrigued a reader, prepared him to explanation of rocket science solution we're going to use soon, and...
just finished here 🐈

File: _docs/technical/protocols/2017-01-13-binary.md [hunk 4]
```diff
@@ -559,48 +573,182 @@ Example:
...
+Currently Cardano supports 3 address types:
+- Public key address (`ATPublicKey`): extended Ed25519 public key as spending
+data, corresponding Ed25519 signature as verification data. 
+- Script address (`ATScript`): Plutus validator script as spending data, Plutus
+redeemer script as verification data.
+- Redeem address (`ATRedeem`): plain Ed25519 public key as spending data,
+  Ed25519 signature as verification data. Used only in genesis to provide
+  pre-sold ADA to holders.
+
+Also there is special `ATUnknown` address type added for backward compatibility.
```
Comment (Martoon-00) [inline _docs/technical/protocols/2017-01-13-binary.md]: Not forward compatibility?

**Review by Martoon-00 — CHANGES_REQUESTED** [review]
Please note my minor comments

**Comment by adinapoli-iohk** [comment]
@flyingleafe @Martoon-00 What should we be doing with this? If we think it would take too much time to write it properly (due to the need of getting technical into the binary format) we should close this PR and mark the relevant ticket as postponed or aborted. Can I leave this to you guys?

**Comment by Martoon-00** [comment]
Yep, I think we will deal with it quickly

File: _docs/technical/protocols/2017-01-13-binary.md (hunk 3 above)
Comment (flyingleafe) [reply _docs/technical/protocols/2017-01-13-binary.md:651]: I added couple of words about Shelley era in the next sentence, but I don't think that this particular file is the right place to describe delegation schemes in detail.

**Review by Martoon-00 — APPROVED** [review]
Kaef

File: _docs/technical/2017-05-14-hd-wallets.md [hunk 5]
```diff
@@ -124,6 +95,34 @@ For wallet to operate over some subtree, one needs to provide either:
...
+
+Let `A(K)` denote the address that holds information about keypair `K`. Let
+`child(K, i)` denote the `i`-th child keypair of `K`. Let `tree(K)` denote the
+tree of addresses for keypairs, derived from `K` (and having positive balance)
+and held in **utxo**.
+
+`a -> b` denotes `b` is derivable form `a`. `a -x b` denotes `b` is not
+derivable from `a` (under no circumstances):
+
+    priv(K) -> pub(K)
```
Comment (pva701) [inline _docs/technical/2017-05-14-hd-wallets.md:108]: @flyingleafe priv))000) :wave:

**Pull request merged**